### PR TITLE
[algo] fix: normalize critic value loss over the global mini-batch, not per micro-batch

### PR DESCRIPTION
**Pull request by abtonmoy**
### What does this PR do?

The critic value loss is normalized per micro-batch instead of per global mini-batch, so the accumulated critic gradient is inflated by the number of micro-batches and changes when the micro-batch count changes (for example when toggling `use_dynamic_bsz`). This effectively scales the critic learning rate by a config-dependent factor. It affects classic PPO runs with a value model (`adv_estimator=gae`, `use_critic=true`) on both the FSDP and Megatron backends.

The actor loss already handles this: `ppo_loss` forwards the global batch info (`dp_size`, `batch_num_tokens`, `global_batch_size`, `loss_scale_factor`) into `agg_loss`, so its summed-over-micro-batch gradient equals the global token-mean and is invariant to how the mini-batch is split. `value_loss` does not forward that info, so `agg_loss` falls back to the local micro-batch token count (`loss_mask.sum()`). This PR makes `value_loss` forward the same info.

The gap was introduced in the engine refactor (#4211): `ppo_loss` gained `global_batch_info` in that change while `value_loss` did not.

### Root cause

`value_loss` -> `compute_value_loss` -> `agg_loss` was called without the global normalization info. For the default `token-mean` mode, `agg_loss` then normalizes by the local micro-batch token count and multiplies by `dp_size=1`, so each micro-batch is self-normalized. Because gradient accumulation sums the per-micro-batch losses, the accumulated gradient is the sum of per-micro-batch means, i.e. about `K` times the correct global token-mean for `K` micro-batches. Reproduced with the real `agg_loss` (whole mini-batch vs. accumulation over `K` micro-batches):

```
correct global token-mean = 0.515
K=1: actor 0.515 (1.00x) | critic 0.515 (1.00x)
K=2: actor 0.515 (1.00x) | critic 1.055 (2.05x)
K=4: actor 0.515 (1.00x) | critic 2.094 (4.06x)
```

### Changes

- `verl/workers/utils/losses.py`: `value_loss` now forwards the global batch info to `compute_value_loss` directly (`dp_size`, `batch_num_tokens`, `global_batch_size` from the batch, `loss_scale_factor` from config). When global normalization is active the `critic/vf_loss` metric is aggregated with `SUM` (each micro-batch contributes a partial sum), otherwise `MEAN`, so the reported metric stays a correct global-batch mean (`reduce_metrics` averages plain-float metrics across micro-batches).
- `verl/trainer/ppo/core_algos.py`: `compute_value_loss` accepts `dp_size`, `batch_num_tokens`, `global_batch_size`, `loss_scale_factor` (defaults preserve current behavior) and forwards them to `agg_loss`.
- `verl/workers/config/critic.py`: `CriticConfig` gains `loss_scale_factor`, mirroring `ActorConfig`.
- `verl/trainer/config/critic/critic.yaml`: adds `loss_scale_factor`, inheriting from the actor via `oc.select` (same pattern as `loss_agg_mode`); the four `_generated_*.yaml` reference configs are regenerated.
- `tests/trainer/ppo/test_value_loss_normalization_on_cpu.py`: new network-free CPU regression tests (see below).

### Scope and limitations

- The `dp_size` cross-rank FSDP gradient reduction requires multiple GPUs; the CPU tests simulate it arithmetically (split the batch across shards, apply the `dp_size` multiplier, mean over shards) but do not run a real multi-GPU reduction.
- The full engine path (`value_loss` through `no_padding_2_padding` / the FSDP engine) needs flash-attn and the no-padding nested-tensor format, so it is exercised by the project's GPU tests rather than the CPU suite here.
- For the `seq-mean-token-sum-norm` mode, micro-batch invariance requires a constant `loss_scale_factor` when micro-batches are padded to different widths, the same requirement the actor already has.

### Tests

Local verification (CPU only):

- Checks pass on the changed files: `ruff` lint, `ruff-format`, `mypy` (no issues in the three changed source files), `py_compile`, and the Apache license header on the new test file. `scripts/generate_trainer_config.sh` regenerates the four reference configs with no further diff (the only change is the added critic `loss_scale_factor` line).
- `pytest tests/trainer/ppo/test_value_loss_normalization_on_cpu.py`: 26 passed. Coverage: micro-batch invariance across all four `loss_agg_mode` values, variable-length sequences, the `dp_size` FSDP mean-reduction (simulated), the `critic/vf_loss` `SUM` metric path via the real `append_to_dict`/`reduce_metrics`, `loss_scale_factor` forwarding, `vf_clipfrac` invariance, a real-autograd gradient-accumulation check, and a characterization of the pre-fix per-micro-batch inflation.
- `pytest tests/trainer/ppo/test_core_algos_on_cpu.py`: 23 passed (no regression in the existing core-algos suite).
- `pytest tests/trainer/ppo -k on_cpu --ignore=tests/trainer/ppo/v2`: 108 passed. The `tests/trainer/ppo/v2` collection error in the unfiltered run is a pre-existing Megatron import that is unavailable in the CPU environment and unrelated to this change.

The multi-GPU and full-engine coverage noted under Scope is left to the project CI.

### Duplicate check

No open PR addresses the current critic path (`verl/workers/utils/losses.py::value_loss`).

- #5767 described this issue but its diff only touched the legacy `verl/workers/actor/dp_actor.py` / `verl/workers/critic/dp_critic.py`, which were removed in the engine refactor. It was closed with the conclusion that the issue does not exist in the current path; that conclusion concerned the metric-accumulation angle and does not cover the value-loss gradient normalization, which is still present on the live path (see the reproduction above).
- #3863 addresses the same concern via a pre-refactor mechanism (`mini_batch_full_token_count`) on files that no longer exist. It is stale and superseded by the current global-batch normalization approach.

---

**AI assistance disclosure**: This PR was developed with AI assistance. The submitting human has reviewed every changed line and is responsible for the change end to end. Verification is summarized in the Tests section: the lint/type/config checks and the CPU test suite pass locally; multi-GPU and full-engine validation is left to the project CI.

### CODE REVIEW

File: verl/workers/utils/losses.py [hunk 1]
```diff
@@ -171,16 +192,13 @@ def value_loss(config: CriticConfig, model_output, data: TensorDict, dp_group=No
         response_mask=response_mask,
         cliprange_value=config.cliprange_value,
         loss_agg_mode=config.loss_agg_mode,
+        **config.global_batch_info,
     )
```
[the remark this answers is not in the document]
Comment (abtonmoy) [reply verl/workers/utils/losses.py:197]: Done, thanks. Passed the four args directly and dropped the `config.global_batch_info` mutation; also removed the now-unused `global_batch_info` field from `CriticConfig`. The critic has a single consumer of this info, so unlike `ppo_loss` (which shares it across the policy, entropy, and KL terms) there was no reason to bundle it onto the config here.

**Review by wuxibin89 — APPROVED** [review]
